packages/strapi/lib/middlewares/cors/index.js

```diff
@@ -32,8 +32,20 @@ module.exports = strapi => {
 
       strapi.app.use(
         cors({
-          origin: function(ctx) {
-            const whitelist = Array.isArray(origin) ? origin : origin.split(/\s*,\s*/);
+          origin: async function(ctx) {
+            let originList;
+
+            if (typeof origin === 'function') {
+              if (origin.constructor.name === 'AsyncFunction') {
```

Comment:
you don't need this condition you can await the call even if not async :)

Reply:
oh ok cool, will make change and push up changes